audio/AudioIO.cpp

```diff
@@ -31,7 +31,7 @@
 #endif
 #include "../os/windows/AudioInputWASAPI.h"
 #include "../os/windows/AudioOutputWASAPI.h"
-#elif defined(__linux__) || defined(__FreeBSD_kernel__) || defined(__gnu_hurd__)
+#elif defined(__linux__) || defined(__FreeBSD__) || defined(__gnu_hurd__)
```

Comment:
This replacing the `__FreeBSD_kernel__` macro with short `__FreeBSD__` may be incompatible with Debian/kFreeBSD. The kernel macro is defined by compiler likewise.

https://github.com/gcc-mirror/gcc/blob/2ab5c3d5457f0d480c6423ee7ac52ce1f98592c9/gcc/config/kfreebsd-gnu.h#L24

But it does not seem to work in original FreeBSD.

And to be honest, today we have no working build of Telegram Desktop for Debian on FreeBSD. The port was abandoned and had little support.

Reply:
@mymedia2 but Debian/kFreeBSD defines `__linux__`, doesn't it?

Reply:
No, it doesn't.